lib/events/firestoreevents/firestoreevents.go

```diff
@@ -466,84 +446,73 @@ func (l *Log) searchEventsOnce(fromUTC, toUTC time.Time, namespace string, limit
 		Where(createdAtDocProperty, "<=", toUTC.Unix()).
 		OrderBy(createdAtDocProperty, firestoreOrdering)).
 		Limit(limit)
-	if len(filter.eventTypes) > 0 {
-		query = query.Where(eventTypeDocProperty, "in", filter.eventTypes)
-	}
 
 	start := time.Now()
 	docSnaps, err := query.Documents(l.svcContext).GetAll()
 	batchReadLatencies.Observe(time.Since(start).Seconds())
 	batchReadRequests.Inc()
 	if err != nil {
-		return nil, 0, "", firestorebk.ConvertGRPCError(err)
-	}
-
-	// Correctly detecting if you've reached the end of a query in firestore is
-	// tricky since it doesn't set any flag when it finds that there are no further events.
-	// This solution here seems to be the most common, but I haven't been able to find
-	// any documented hard guarantees on firestore not early returning for some reason like response
-	// size like DynamoDB does. In short, this should work in all cases for lack of a better solution.
-	if len(docSnaps) < limit {
-		reachedEnd = true
+		return nil, "", firestorebk.ConvertGRPCError(err)
 	}
 
 	g.WithFields(log.Fields{"duration": time.Since(start)}).Debugf("Query completed.")
 	for _, docSnap := range docSnaps {
 		var e event
 		err = docSnap.DataTo(&e)
 		if err != nil {
-			return nil, 0, "", firestorebk.ConvertGRPCError(err)
+			return nil, "", firestorebk.ConvertGRPCError(err)
 		}
 
 		data := []byte(e.Fields)
-		if totalSize+len(data) >= spaceRemaining {
+		if totalSize+len(data) >= events.MaxEventBytesInResponse {
 			break
 		}
 
 		var fields events.EventFields
 		if err := json.Unmarshal(data, &fields); err != nil {
-			return nil, 0, "", trace.Errorf("failed to unmarshal event %v", err)
+			return nil, "", trace.Errorf("failed to unmarshal event %v", err)
 		}
 
+		time := docSnap.Data()[createdAtDocProperty].(int64)
+		lastKey = strconv.Itoa(int(time)) + ":" + docSnap.Ref.ID
```

Comment:
nit: `Sprintf` seems easier to read to me, feel free to ignore.
```suggestion
		lastKey = fmt.Sprintf("%d:%s", int(time), docSnap.Ref.ID)
```

Reply:
Sometimes but it's a lot slower